go-controller/pkg/node/port_claim.go

```diff
@@ -8,41 +8,58 @@ import (
 
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
 	kapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/klog"
+	utilnet "k8s.io/utils/net"
 )
 
-type handler func(port int32, protocol kapi.Protocol, svc *kapi.Service) error
+type handler func(desc string, ip string, port int32, protocol kapi.Protocol, svc *kapi.Service) error
 
-type localPort interface {
-	open(port int32, protocol kapi.Protocol, svc *kapi.Service) error
-	close(port int32, protocol kapi.Protocol, svc *kapi.Service) error
+type localPortHandler interface {
+	open(desc string, ip string, port int32, protocol kapi.Protocol, svc *kapi.Service) error
+	close(desc string, ip string, port int32, protocol kapi.Protocol, svc *kapi.Service) error
 }
 
-var port localPort
+var portHandler localPortHandler
 
-type activeSocket interface {
-	Close() error
-}
+var portOpener utilnet.PortOpener
 
 type portClaimWatcher struct {
 	recorder          record.EventRecorder
 	activeSocketsLock sync.Mutex
-	activeSockets     map[kapi.Protocol]map[int32]activeSocket
+	localAddrSet      map[string]net.IPNet
+	portsMap          map[utilnet.LocalPort]utilnet.Closeable
 }
 
-func newPortClaimWatcher(recorder record.EventRecorder) localPort {
+// Prefix for a LocalPort description.
+type localHostDescrPrefix string
```

Comment:
nit: is there a reason that you need to define the prefixes as a separate type? If they are eventually just going to be string, might as well keep them that way?

Reply:
Done